### DISCUSSION TITLE
RN expo app: Attempt to connect to relay via `transportOpen` has stalled. Retrying...

### DISCUSSION THREAD

**Discussion by RobertoSnap** [WalletConnect/walletconnect-monorepo#2835] [Developer Support] [answered]
Suddenly I'm getting this when trying to init a SignClient from React Native expo build.

![CleanShot 2023-06-28 at 13 08 55](https://github.com/WalletConnect/walletconnect-monorepo/assets/4772930/3eba70bb-499d-4235-9449-9eaae5d0cd99)

Package.json
```json
  "dependencies": {
    "expo": "~48.0.18",
    "expo-status-bar": "~1.4.4",
    "react": "18.2.0",
    "react-native": "0.71.8",
    "@walletconnect/react-native-compat": "^2.8.4",
    "@walletconnect/sign-client": "^2.8.4",
    "@walletconnect/types": "^2.8.4",
    "@babel/plugin-proposal-class-properties": "^7.18.6",
    "@babel/plugin-proposal-private-methods": "^7.18.6",
    "@react-native-async-storage/async-storage": "1.17.11",
    "ethers": "^6.6.1",
    "react-native-get-random-values": "~1.8.0",
    "zustand": "^4.3.8"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@types/react": "~18.0.14",
    "typescript": "^4.9.4"
  },
  ```

**Comment by finessevanes** [comment]
Hey there, thanks for reporting. Would you mind including:
1) Reproducible code via a codeBox, error logs, and/or reproduction steps

**Reply by RobertoSnap** [reply]
EDIT: Works now, but i think the error will reapper. Il try to catch whats going on.

[https://github.com/RobertoSnap/swip-app-2](https://github.com/RobertoSnap/swip-app-2)

#### on pc
git clone https://github.com/RobertoSnap/swip-app-2
cd swip-app-2
yarn
yarn start

#### on phone
Install expo go
scan qr code
open app

Now it works. 
I think this will happen again, but il report back then.

Thanks for quik reply.

**Reply by ramvi** [reply]
I am encountering the same problem as you. However, despite following the steps you've outlined, the issue still persists on my end. 

I cloned the repository from https://github.com/RobertoSnap/swip-app-2 into a new directory (swip-app-3), navigated to the swip-app-3 directory, installed all dependencies using `yarn`, and started the application with `yarn start`.

For the mobile part, I installed Expo Go on my phone and scanned the QR code, but when I try to open the app, it doesn't work as expected.

@finessevanes 
Is this something you can help troubleshoot?

Thanks for your help and your quick responses.

**Reply by RobertoSnap** [reply]
This is still happening for some but not others:

I see its the same error here: https://github.com/orgs/WalletConnect/discussions/2793 but dont think .com is the problem.

**Reply by RobertoSnap** [reply]
We have not figured out why yet, though running the app in Web is a workaround while developing. 

So probably something interfering in the websockets between expo, app and walletconnect

**Reply by RobertoSnap** [reply]
Cant find similear react native issus online so adding more details here:

This is what we see in debugger:
![CleanShot 2023-06-29 at 23 00 12](https://github.com/WalletConnect/walletconnect-monorepo/assets/4772930/0d8350f8-e675-408d-875d-41775f67899e)

As text:
`Socket stalled when trying to connect to wss://relay.walletconnect.org`

```bash
jsEngine: "hermes"
message: "Socket stalled when trying to connect to wss://relay.walletconnect.org"
stack: "Error: Socket stalled when trying to connect to wss://relay.walletconnect.org\n    at anonymous (http://192.168.68.53:19000/node_modules/expo/AppEntry.bundle//&pl
``

**Reply by RobertoSnap** [reply]
Another observation: 

When this happens, not logs are displayed in terminal running expo. I assume that is sent by Websockets, so something here.

**Reply by RobertoSnap** [reply]
if i remove these plugins from babel.config.js

```
      '@babel/plugin-proposal-class-properties',
      '@babel/plugin-proposal-private-methods',
```
And remove anything with WalletConnect (because it fails without them). Then i see normal Websocket is restored.

**Reply by RobertoSnap** [reply, marked as answer]
These plugins seems to cause the issue, ref: https://github.com/facebook/metro/issues/877

` '@babel/plugin-proposal-class-properties',
      '@babel/plugin-proposal-private-methods',`

I added them because we use Ethers, so i removed Ethers and these plugins and Wallet connect websocket worked:

**Reply by RobertoSnap** [reply]
Fixed by downgrading ethers and adding shims for it instead.

**Reply by ramvi** [reply]
Dear @RobertoSnap,

I can't thank you enough for your dedication. Your efforts have been nothing short of extraordinary! You've proven to be a real beacon of light in times of need, illuminating our way with your resourcefulness and tenacity. You truly are a star! Not just any star, but the brightest one in our sky. Your expertise and commitment are unmatched, and for that, we are immensely grateful. To say you are an angel wouldn't quite cover it; you're more like a saving angel, swooping in with solutions when all seemed lost.

**Reply by linghuccc** [reply]
But I need to use Ethers in my project... So what to do if Ethers is needed?
